cami.c

```diff
@@ -438,18 +484,35 @@ struct ami_session *ami_connect(const char *hostname, int port, void (*callback)
 		pthread_mutexattr_destroy(&attr);
 	}
 
-	if (pthread_create(&ami->ami_thread, NULL, ami_loop, ami)) {
-		ami_error(ami, "Unable to create AMI thread: %s\n", strerror(errno));
-		ami_cleanup(ami);
-		goto cleanup;
+	{
+		pthread_attr_t attr;
+		pthread_attr_init(&attr);
+		pthread_attr_setstacksize(&attr, 2 * 1024 * 1024);
+		ret = pthread_create(&ami->ami_thread, &attr, ami_loop, ami);
```

Comment:
I'm assuming Mac OS doesn't like a NULL 2nd argument to pthread_create, and that's the reason for this?
If so, maybe a comment to that effect would be helpful

Reply:
I've written code that passes NULL and worked just fine.  But, while testing "this" code I kept getting a BUS error.  Passing a "attr", something I have also used in my code, resolved the issue.  For now, I'm hesitant to say that this needed for macOS.